yb-voyager/src/callhome/diagnostics.go

```diff
@@ -349,8 +353,44 @@ func SendPayload(payload *Payload) error {
 
 // We want to ensure that no user-specific information is sent to the call-home service.
 // Therefore, we only send the segment of the error message before the first ":" as that is the generic error message.
-// Note: This is a temporary solution. A better solution would be to have
-// properly structured errors and only send the generic error message to callhome.
-func SanitizeErrorMsg(errorMsg string) string {
-	return strings.Split(errorMsg, ":")[0]
+// Accepts error type, returns empty string if error is nil.
+func SanitizeErrorMsg(err error) string {
+	if err == nil {
+		return ""
+	}
+	errorMsg := strings.Split(err.Error(), ":")[0]
+	additionalContext := getSpecificContextForError(err)
+	if additionalContext != nil {
+		var contextData string
+		contextJsonBytes, err := json.Marshal(additionalContext)
+		if err != nil {
+			// should not happen.
+			log.Errorf("callhome: error while marshalling additional context for error: %v", err)
+		}
+		contextData = string(contextJsonBytes)
+		errorMsg = fmt.Sprintf("%s: %s", errorMsg, contextData)
+	}
+	return errorMsg
+}
+
+func getSpecificContextForError(err error) map[string]string {
```

Comment:
`getSpecificNonSensitiveContextForError` ?
Just to enforce and avoid accidental sending of sensitive info